thirdparty/hnswlib/hnswlib/neighbor.h

```diff
@@ -22,18 +22,29 @@ struct Neighbor {
     operator<(const Neighbor& other) const {
         return distance < other.distance;
     }
+    inline bool
+    operator>(const Neighbor& other) const {
+        return distance > other.distance;
+    }
 };
-
+typedef std::priority_queue<Neighbor, std::vector<Neighbor>, std::greater<Neighbor>> IteratorMinHeap;
 class NeighborSet {
  public:
     explicit NeighborSet(size_t capacity = 0) : capacity_(capacity), data_(capacity_ + 1) {
     }
 
+    // will push any neighbor that does not fit into NeighborSet to disqualified.
```

Comment:
this is for #75 as well. When searching for iterator, those points removed from NeighborSet may be qualified candidates as the iterator iterates, thus we need to collect instead of disposing them. Will add comment.